### ISSUE TITLE
🤔 targetWorkspaces still a required value & wildcardName providing decoding error 

### ISSUE THREAD

**Issue by rafeedm** [hashicorp/terraform-cloud-operator#324]
### Operator Version, Kind and Kubernetes Version
 - Operator version: 2.2.0
 - Kind: AgentPool
 - Kubernetes version: v1.25.6

### YAML Manifest File
```yaml
targetWorkspaces:
    - wildcardName: appprj*
```

### Output Log
When trying to apply agent pool manifest without `targetWorkspaces` I get the error: 
```
The AgentPool "abc-pool" is invalid: spec.autoscaling.targetWorkspaces: Required value
```

When trying to use wildcards, like the above manifest, I get the error:
``` 
Error from server (BadRequest): error when creating "agentpool.yaml": 
AgentPool in version "v1alpha2" cannot be handled as a AgentPool: 
strict decoding error: unknown field "spec.autoscaling.targetWorkspaces[0].wildcardName"
```

### Question
I followed the docs step by step but wondering if I'm missing something which is causing the above errors. Note when declaring the workspace name explicitly the autoscaling behaviour works as expected, so I assume the creation of secrets, and the AgentPool provisioning was done correctly. I used helm to deploy the operators, v2.2.0, can someone provide some insights as to why these features aren't working? Without them the operator would be unusable for our use case

### References
Both optional Target Workspace and wildcard name searches are in release [v2.0.0](https://github.com/hashicorp/terraform-cloud-operator/releases/tag/v2.0.0) in the following PR:
 - [GH-274](https://github.com/hashicorp/terraform-cloud-operator/pull/274)

### Community Note

* Please do not leave "+1" or other comments that do not add relevant new information or questions, they generate extra noise for issue followers and do not help prioritize the request.
* Please vote on this issue by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request.
* If you are interested in working on this issue or have submitted a pull request, please leave a comment.

**Comment by arybolovlev** [comment]
Hi @rafeedm,

Could you please [share](https://gist.github.com/) the Agent Pool CRD from your cluster?

```console
$ kubectl get crd agentpools.app.terraform.io -o yaml
```

I suspect that you might have had a beta version installed earlier and once you installed a new one, the CRD remained unchanged.

Thanks.

**Comment by rafeedm** [comment]
Hey @arybolovlev 

Here's the [link](https://gist.github.com/rafeedm/2fc5293b429d82f6cb7ca2081a0c4e25)

**Comment by rafeedm** [comment]
Hey @arybolovlev 

Just pinging again, wondering if you or anyone on the team has gotten a chance to look into this

**Comment by arybolovlev** [comment]
Hi @rafeedm,

It looks like the CRD that you have shared is from a pre-GA installation where `targetWorkspaces` was marked as required and  `wildcardName` didn't exist.

The thing is that [Helm does not upgrade CRDs](https://helm.sh/docs/chart_best_practices/custom_resource_definitions/#some-caveats-and-explanations) when the upgrade operation is performed. That is the reason why your CRD remained the same after upgrading the version of the operator.

The solution here would be to use `kubectl replacement` to install a new CRD that correlates with the version of the operator that you have installed.

```console
$ kubectl replace \
    -f https://raw.githubusercontent.com/hashicorp/terraform-cloud-operator/v2.2.0/charts/terraform-cloud-operator/crds/app.terraform.io_agentpools.yaml
```

In `2.2.0` we also introduced a new controller `Project` and if you upgrade the Operator from any previous versions via Helm, it will not install a new controller CRD for the same reason as above. This will lead to the situation when the Operator pods are constantly restarting with the following message in the logs:

```
2024-01-24T07:44:14Z	ERROR	controller-runtime.source.EventHandler	if kind is a CRD, it should be installed before calling Start	{"kind": "Project.app.terraform.io", "error": "no matches for kind \"Project\" in version \"app.terraform.io/v1alpha2\""}
```

In this case, you need to install a new CRD manually via `kubectl apply`:

```console
$ kubectl apply \
    -f https://raw.githubusercontent.com/hashicorp/terraform-cloud-operator/v2.2.0/charts/terraform-cloud-operator/crds/app.terraform.io_projects.yaml
```

I will update the [FAQ](https://github.com/hashicorp/terraform-cloud-operator/blob/main/docs/faq.md) and Helm chart [README](https://github.com/hashicorp/terraform-cloud-operator/blob/main/charts/terraform-cloud-operator/README.md) with the same explanation and instructions.

I hope that helps.

**Comment by rafeedm** [comment]
Hey @arybolovlev 

Thanks for the help, I was able to resolve the issue. It may also help to include some other items on the FAQ:

- Deleting agentpools by simply running `kubectl delete agentpools.app.terraform.io name-of-pool` may get stuck. I don't know if this was the proper way, but I had to remove the finalizer from the agentpool CRD and then delete the CRD, then the pools were able to be deleted
- You mentioned helm upgrade issue, but helm uninstall also won't be able to handle the CRDs, so if someone wants to start from a clean slate again they can't just uninstall the helm release and install again, the CRDs will remain unchanged. 
- Due to not being able to delete agentpools, when I tried to `kubectl apply -f agentpool.yaml` with new changes for an existing pool, it was marked for termination but did not actually update. I have not tested, but perhaps `replace` or `edit` may be better options
